From 5d06625b55f0c48a239b594475e544ceb0fc5596 Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Fri, 7 Aug 2020 11:57:23 +0530
Subject: [PATCH 32/32] Add instructive comments and clarify kernel message
 indicating the CPU variant

- U-Boot overwrites the dummy size in the DDR3 memory node with the actual
  value, don't bother editing it.
- Comment that wdog1 needs to be explicitly disabled and that watchdog uses
  WDOG2_B output: Wdog2 was meant as TrustZone watchdog and doesn't connect
  to the System Reset Controller (SRC), but we need an external reset path
  via -->BMC-->POR to reset module and carrier devices anyway, and
  pin-muxing only left WDOG2_B unoccupied. And so far we're not interested
  in TrustZone.
- The kernel code mostly talks about i.MX6 "Quad" and "DualLite", but they
  actually mean "Quad/Dual" and "DualLite/Solo".

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi | 5 ++++-
 arch/arm/mach-imx/mach-imx6q.c  | 2 +-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index c25a1d484cfd..f46488891023 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -40,7 +40,7 @@
 
 	memory: memory@10000000 {
 		device_type = "memory";
-		reg = <0x10000000 0x40000000>;
+		reg = <0x10000000 0x40000000>;	/* dummy size, overwritten by U-Boot */
 	};
 
 	reg_smarc_lcd_vdd: regulator-smarc-lcd-vdd {
@@ -1270,10 +1270,13 @@
 };
 
 &wdog1 {
+	/* Contrary to other nodes Freescale's dtsi doesn't default to "disabled" here. */
 	status = "disabled";
 };
 
 &wdog2 {
+	/* This is the TrustZone watchdog, however pin mux leaves no other choice.
+	 * We don't need an SoC reset (wdog_rst_b), but WDOG2_B --> system wide POR. */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_wdog>;
 	fsl,ext-reset-output;
diff --git a/arch/arm/mach-imx/mach-imx6q.c b/arch/arm/mach-imx/mach-imx6q.c
index d6d84d7aaf1d..2ef65e758f13 100644
--- a/arch/arm/mach-imx/mach-imx6q.c
+++ b/arch/arm/mach-imx/mach-imx6q.c
@@ -346,7 +346,7 @@ static void __init imx6q_init_machine(void)
 	if (cpu_is_imx6q() && imx_get_soc_revision() >= IMX_CHIP_REVISION_2_0)
 		imx_print_silicon_rev("i.MX6QP", IMX_CHIP_REVISION_1_0);
 	else
-		imx_print_silicon_rev(cpu_is_imx6dl() ? "i.MX6DL" : "i.MX6Q",
+		imx_print_silicon_rev(cpu_is_imx6dl() ? "i.MX6DL/S" : "i.MX6Q/D",
 				imx_get_soc_revision());
 
 	parent = imx_soc_device_init();
-- 
2.17.1

